src/Microsoft.OData.Core/Metadata/EdmLibraryExtensions.cs

```diff
@@ -1868,6 +1982,19 @@ private static int InheritanceLevelFromSpecifiedInheritedType(this IEdmStructure
 
             return inheritanceLevelsFromBase;
         }
+
+        private static void EnsureOperationBoundWithBindingParameter(this IEdmOperation operation)
+        {
+            if (!operation.IsBound)

```

Comment:
Make an assert instead.

Reply:
Converted to assert for functionoverload resolver but did not revert for select segment code path. 

---
In reply to: [370876004](https://github.com/OData/odata.net/pull/1602#discussion_r370876004) [](ancestors = 370876004)